bin/ruby-build

```diff
@@ -8,6 +8,7 @@
 #   --definitions   List all local definitions, including outdated ones
 #   --version       Show version of ruby-build
 #
+#   -d, --dir       Install the Ruby in <prefix>/<definition> instead of <prefix>
```

Comment:
My nitpick is that maybe this flag should receive a value instead of being a boolean flag.

When I see an invocation like this:
```
ruby-build 3.2.1 --dir ~/.rubies
```
it would seem to me like the `--dir` flag is passed a value of `~/.rubies`, but that's not how this is implemented right now. This would likely work: `ruby-build --dir 3.2.1 ~/.rubies`, and so would this: `ruby-build 3.2.1 ~/.rubies --dir`, but I do not think these invocations should work because they look "wrong" and I believe that they should error out. WDYT?

Reply:
I tried to fit the existing usage `ruby-build [-kpv] <definition> <prefix>`.
If the prefix becomes optional it feels like opening Pandora's box, so that's why I preferred an option without a value. Also it seems currently ruby-build doesn't have any option taking an argument, and I would like to avoid having to add that.

I think all 3 variants you showed are fine to use.
I used `--dir` in the example because it works and looks cool, but `ruby-build --dir 3.2.1 ~/.rubies` is just as fine.

I'm happy to update the PR's description if that helps.

Reply:
Actually I should update the usage line to `ruby-build [-kpvd]` and then the documented way to use this is `ruby-build -d definition prefix`.

Reply:
I updated the usage line and the PR descriptions to use `ruby-build -d`.
The fact `-d` in the middle works is then like an easter egg.